Add build script for OSX and .pkg installer

darcyliu opened the pull request:
Solve the issue https://github.com/mlpack/mlpack/issues/364

Commit:
Add script for building OSX .pkg installer

Building an OSX installer package
1. Compiling the source code
2. Run the ’build-osx-pkg.sh’ script

It will generate a package file ‘mlpack.pkg’ without code sign.

Commit:
Add build script for OSX

zoq commented on packaging/distribution.plist:
```diff
@@ -0,0 +1,26 @@
+        <line choice="install"/>
+    </choices-outline>
+    
+    <choice id="install" visible="true" title="Install Rust" description="Install the Rust compiler, package manager and documentation."
```
I think, we should write "Install mlpack".

zoq commented on packaging/distribution.plist:
```diff
@@ -0,0 +1,26 @@
+    <options customize="never" require-scripts="false"/>
+    <!-- <welcome file="README.md" mime-type="text/html" /> -->
+    <license file="LICENSE.txt" mime-type="text/html" />
+    <conclusion file="conclusion.txt" mime-type="text/html" />
```
I'm not quite sure, if this conclusion.txt is autogenerated?

zoq commented:
Thanks for taking the time, to setup the installer for OSX.

Just some notes: Once we configured the jenkins OSX build, there is no need to write an extra file to build lapack. We just have to make sure, that all the necessary files are in the right directory.

Commit:
Update installer description

This file is adapted from https://github.com/rust-lang/rust-installer

The command-line programs will be installed in the /usr/local/bin, the
library will be placed in /usr/local/lib and the header files can be
found at /usr/local/include

Commit:
Update the link for CI server

darcyliu commented:
Sorry for the mistakes.
It is true that we do not need compile lapack, armadillo and boost every time. I include these steps in the script just for reminder.

Could we use the address http://ci.mlpack.org/ in the documentation and make a redirection to the real server, then if the server location is changed, we only need change the redirection without modify the documentation everywhere.
